.github/workflows/build-and-deploy.yml

```diff
@@ -4,20 +4,19 @@ jobs:
   build-and-deploy:
     name: Build and Deploy
     runs-on: ubuntu-latest
+    # Only run on PRs if the source branch is on someone else's repo
+    if: ${{ github.event_name != 'pull_request' || github.repository != github.event.pull_request.head.repo.full_name }}     
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
         with:
           persist-credentials: false
 
       - name: Cache Node Modules
-        uses: actions/cache@v1
+        uses: actions/cache@v3
         with:
           path: ~/.npm
           key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
-          restore-keys: |
-            ${{ runner.os }}-node-
-            ${{ runner.os }}-
```

Comment:
Out of curiosity, why did you remove this?

Reply:
Looking into actions/cache's docs, it isn't needed.

Reply:
I think `${{ runner.os }}-` might indeed not be needed, but `${{ runner.os }}-node-` sounds useful to me in case `package-lock.json` is updated, so it can reuse most of the old cached packages.

The [example](https://github.com/actions/cache/blob/main/examples.md#node---npm) in their repository also includes that. Or to which docs were you referring?

Reply:
Alright, reverted